### ISSUE TITLE
No math formaula rendered after enabled webkit-rendering

### ISSUE THREAD

**Issue by CanYellow** [retext-project/retext#576]
Hi, thank you for your work.
I installed retext on manjaro via pip3, and installed mathjax in the right directory. But I still can't get any formula rendering output in the live preview window. I thought It might be some wrong config or conflict about qtwebengine, here is some qt package install on my system
```
extra/qt5-wayland 5.15.2+kde+r46-2 (qt qt5) [installed]
extra/qt5-webchannel 5.15.2+kde+r5-2 (qt qt5) [installed]
extra/qt5-webengine 5.15.8-7 (qt qt5) [installed]
extra/qt5-webkit 5.212.0alpha4-11 [installed]
extra/qt5-x11extras 5.15.2-2 (qt qt5) [installed]
```
could you give me some suggestions.? thanks in advance!

More info
I read the html code of md file. it said "<script type="text/javascript" src="file:///usr/share/mathjax/tex-chtml.js" async>", and I am totally sure that I have the right mathjax (tex-chtml.js) path. maybe the [wikipage](https://github.com/retext-project/retext/wiki/Math) has some tiny error on mathjax path.

I tried launch retext from terminal, there was no any error output.

**Comment by CanYellow** [comment]
Actually, the button of webkit-render in edit menu is available and I ticked it.

I already tried install `mathjax2` package by manjaro packager manager `pacman`, and it didn't work. But I forget check whether the `mathjax2` is right or not. I will try to install `mathjax2` to right path manually according  to your suggestion after hous later.

please wait some hours, I will feedback.
 thank you for response!

-----
Yeah. I read the issue list and the webkit project page. and already knew the reason why you removed the webkit supprt. removing was neccessary and It was not a bad idea, I thought.

**Comment by CanYellow** [comment]
I tried the sulotions above.

I  reinstalled retext with pip, and installed mathjax2 with pacman on manjaro. and marked webkit-render. but the problem still exist.

as a second way, I uninstalled the mathjax2 installed with pacman, but reinstalled mathjax2 with pip. the same problem was there unfortunately.

-----
I just can use retext to edit md file. and I think live preview is not very important for me, but if neccessary I can use firefox plugin to do that. so it will be ok if the problem didn't solve.

-----
but if you need me try and test some thing, I am will to doing this.

**Comment by mitya57** [comment]
Can you try the version from master? You can simply clone the repository and run `./retext.py` from there, or install it with `pip install 'git+https://github.com/retext-project/retext.git#egg=ReText'`.

This version should use Qt WebEngine renderer which is superior to Qt WebKit one and does not have any issues.

If it still happens with version from master, please open “Edit → View HTML code” dialog and copy-paste the code from there (or attach it).

**Comment by CanYellow** [comment]
Ok. Thanks! I will try it later today.

**Comment by CanYellow** [comment]
I tried the solution. 
1. I installed retext by `pip` using the command `pip install 'git+https://github.com/retext-project/retext.git#egg=ReText'`.
2. I also tried `git clone` the master repo to get `retext.py` and run it.

with all above two methods I can successfully install retext 7.3.0, that means the terminal outputs `ReText 7.3.0 (Git) ` when typeing `retext --version`.

But with all the two installations. the button `use webkit(chromium) render` is blocked. I can't check it to enable it.

I tried to manually edit the config file to add `useWebEngine=true` option, but it seems the option is setting from retext internal and after launching retext, the config file will re-generate. so it's vast to manually and the option.

I  am sure I have the right pyqt lib as
```
pip install pyqt5 
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: pyqt5 in /usr/lib/python3.10/site-packages (5.15.6)
Requirement already satisfied: PyQt5-sip<13,>=12.8 in /usr/lib/python3.10/site-packages (from pyqt5) (12.9.1)
```
So how can I get the `use webengine(chromium) render` function enable?

-------
also here is the `view html code` output. I think the code generated is OK and the mathjax directory is right, because I used firefox to open the output html code. and it was rendered as I wish.
here is the output html code by check `view html code` button:
```
<!doctype html>
<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<meta name="generator" content="ReText 7.3.0 (Git)">
<title>test</title>

<script>
MathJax = {
  options: {
    renderActions: {
      find: [10, function (doc) {
        for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
          const display = !!node.type.match(/; *mode=display/);
          const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
          const text = document.createTextNode('');
          node.parentNode.replaceChild(text, node);
          math.start = {node: text, delim: '', n: 0};
          math.end = {node: text, delim: '', n: 0};
          doc.math.push(math);
        }
      }, '']
    }
  }
};
</script>
<script type="text/javascript" src="file:///usr/share/mathjax/tex-chtml.js" async></script></head>
<body>
<p>tes</p>
<p>
<script type="math/tex; mode=display">1-1=2</script>
</p>
<p>gor t</p>
<p>
<script type="math/tex; mode=display">
1+1=2
</script>
</p>

</body>
</html>
```

**Comment by mitya57** [comment]
Sorry for the late response. I forgot one important thing: please `pip install PyQtWebEngine`.

**Comment by CanYellow** [comment]
You are welcome.

I didn't check my QtWebEngine because I thought I have installed it just as what I quoted in the initial question description.

After read your response, I checked my installation enviroment. Here it is 
```shell
$ pacman -Qs |grep qt  
local/python-pyqt5 5.15.6-7.1 (pyqt5)
local/python-pyqt5-sip 12.9.1-1
local/qt5-webchannel 5.15.2+kde+r5-2 (qt qt5)
local/qt5-webengine 5.15.8-7 (qt qt5)
local/qt5-webkit 5.212.0alpha4-11
local/qt5-websockets 5.15.2+kde+r7-2 (qt qt5)
......
# I delete other unrelated package from the output.
```
and
```shell
$ pip show
PyQt5                5.15.6
PyQt5-sip            12.9.1
PyQt6-Qt6            6.2.3
PyQt6-sip            13.2.1
......
```
So I installed `PyQtWebEngine` using `pip install PyQtWebEngine`. now the package list was
```shell
$ pip show
PyQt5                5.15.6
PyQt5-sip            12.9.1
PyQt6-Qt6            6.2.3
PyQt6-sip            13.2.1
PyQtWebEngine        5.15.5
PyQtWebEngine-Qt5    5.15.2
```
and re-installed retext with `pip install 'git+https://github.com/retext-project/retext.git#egg=ReText'`

But the problem still existed. I mean nothing changed. Was there something wrong I did?

-------
Some additional information:
1. `PyQtWebEngine` installation info
```shell
$ pip show PyQtWebEngine 
Name: PyQtWebEngine
Version: 5.15.5
Summary: Python bindings for the Qt WebEngine framework
Home-page: https://www.riverbankcomputing.com/software/pyqtwebengine/
Author: Riverbank Computing Limited
Author-email: info@riverbankcomputing.com
License: GPL v3
Location: /home/brighthuang/.local/lib/python3.10/site-packages
Requires: PyQt5, PyQtWebEngine-Qt5, PyQt5-sip
```
2. when installing retext, the error was
```shell
  ERROR: Failed building wheel for ReText
  Running setup.py clean for ReText
Failed to build ReText
Installing collected packages: ReText
    Running setup.py install for ReText ... done
  DEPRECATION: ReText was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. pip 21.0 will remove support for this functionality. A possible replacement is to fix the wheel build issue reported above. You can find discussion regarding this at https://github.com/pypa/pip/issues/8368.
```
I thought that was a unimportant error after I read the GitHub issue page 

3. I also launched retext from linux shell use command: `retext`, just one error when launching retext, I thought the error was not related to my problem. it was
```shell
Using configuration file: /home/brighthuang/.config/ReText project/ReText.conf
Qt: Session management error: networkIdsList argument is NULL
```

**Comment by mitya57** [comment]
Is the “Use Qt WebEngine (Chromium) renderer” option still grayed out? If not, have you enabled it?

**Comment by CanYellow** [comment]
Yes， it's still grayed out. So I can not enable it.

**Comment by mitya57** [comment]
Can you run a Python interpreter and paste these two lines?

```python
from PyQt5.QtCore import QCoreApplication
from PyQt5.QtWebEngineWidgets import QWebEnginePage, QWebEngineView, QWebEngineSettings
```

If the import succeeds then the option should not be grayed out. Please make sure you are running ReText with the same interpreter as you used here.

If the import doesn't succeed, then it's a problem with your Python environment. Maybe distro packages conflict with packages installed via pip? Can you try installing python-pyqt5-webengine with pacman?

**Comment by CanYellow** [comment]
OK, copy that. Thanks for your response. I will try it later today.

**Comment by CanYellow** [comment]
Thanks for you suggestion. 
typed what you said, I get a import error '... undefined symbol: _ZNSt12out_of_rangeC1EPKc... '

I googled it, and found the solution in https://stackoverflow.com/questions/68798645/import-error-with-pyqt5-undefined-symbol-znst12out-of-rangec1epkc-versionqt-5

I uninstalled pyqt5 with pip and reinstalled pyqt5 with pacman and the problem solved.

Now retext can render math formula  in preview/ live-preview window.

Thank you very much!
after what you said,
